src/enclave/rpc_context.h

```diff
@@ -103,6 +103,8 @@ namespace enclave
     {}
   };
 
+  using PathParams = std::map<std::string, std::string>;
```

Comment:
Are params already URL-decoded?

Reply:
No. Currently we only URL-decode (with a function called `url_unescape` because I uhhhhh, brain stumbled) the query. So `/users/Bob%20Bobson` will match `/users/{id}` with `id = "Bob%20Bobson"`. I'll add a decode of the whole path, and then they'll get `id = "Bob Bobson"`

Reply:
After further discussion, we've decided not to decode the whole path. This may introduce awkward null bytes in the things we're still processing as strings. Instead we pass around URL-encoded strings for as long as possible, and its up to the handler to URL-decode things that may be encoded if and when they care about it.